website/docs/cdktf/test/unit-tests.mdx

```diff
@@ -47,9 +55,127 @@ describe("Unit testing using assertions", () => {
 });
 ```
 
+```python
+import pytest
+from cdktf import Testing
+from imports.docker import Image, Container
+from app import MyApplicationsAbstraction # Could be a class extending from cdktf.Resource
+
+class TestApplication:
+
+    stack = TerraformStack(Testing.app(), "stack")
+    app_abstraction = MyApplicationsAbstraction(stack, "app-abstraction")
+    synthesized = Testing.synth(stack)
+
+  def test_should_contain_container(self):
+    assert Testing.to_have_resource(self.synthesized, Container.TF_RESOURCE_TYPE).pass_
+
+  def test_should_use_an_ubuntu_image(self):
+    assert Testing.to_have_resource_with_properties(self.synthesized, Image.TF_RESOURCE_TYPE, {
+            "name": "ubuntu:latest",
+    }).pass_
+```
+
+```java
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import com.hashicorp.cdktf.Testing;
+import imports.docker.Container;
+import imports.docker.Image;
+import com.hashicorp.cdktf.MyApplicationsAbstraction; // Could be a class extending from cdktf.Resource
+
+public class TestApplication {
+
+  private final TerraformStack stack = new TerraformStack(Testing.app(), "stack");
+  private final MyApplicationsAbstraction appAbstraction = new MyApplicationsAbstraction(stack, "resource");
+  private final String synthesized = Testing.synth(stack);
+
+  @Test
+  void shouldContainContainer() {
+    assertTrue(Testing.toHaveResource(synthesized, Container.TF_RESOURCE_TYPE).getPass());
+  }
+
+  @Test
+  void shouldUseUbuntuImage() {
+    assertTrue(Testing.toHaveResourceWithProperties(synthesized, Image.TF_RESOURCE_TYPE, new HashMap<String, Object>() {
+                {
+                    put("name", "ubuntu:latest");
+                }
+    }).getPass());
+  }
+}
+```
+
+```csharp
+using Xunit;
+using HashiCorp.Cdktf; // MyApplicationsAbstraction - Could be a class extending from cdktf.Resource
```

Comment:
```suggestion
using HashiCorp.Cdktf; // MyApplicationsAbstraction - Could be a class extending from Construct
```
I think we recommend Construct now (we used to recomment extending Resource though)
What do you think @DanielMSchmidt?

Reply:
Yeah, I agree we recommend Construct